rustls/src/crypto/mod.rs

```diff
@@ -588,6 +570,47 @@ pub fn default_fips_provider() -> CryptoProvider {
     aws_lc_rs::default_provider()
 }
 
+#[cfg(feature = "defaultproviderenabled")]
+mod crypto_default_provider {
+    use alloc::sync::Arc;
```

Comment:
I have already proposed this update in PR #2089. I am planning to update this cfg attribute to get rid of the new feature options.